examples/mrc/run_mrc.py

```diff
@@ -440,6 +465,38 @@ def compute_metrics(p: EvalPredictionWithProcessing):
         trainer.log_metrics("eval", metrics)
         trainer.save_metrics("eval", metrics)
 
+    if task_args.do_boolean:
+        logger.info("Processing of boolean questions")
+        if not os.path.exists(os.path.join(training_args.output_dir,"eval_predictions.json")):
+            raise Exception(f"No MRC predictions were found at {training_args.output_dir}")
+        with open(task_args.boolean_config, 'r') as f:
+            boolean_config = json.load(f)
+
+        boolean_config['qtc']['output_dir'] = training_args.output_dir+"/qtc"
+        boolean_config['qtc']['test_file'] = training_args.output_dir + "/eval_predictions.json"
+        boolean_config['evc']['output_dir'] = training_args.output_dir+"/evc"
+        boolean_config['evc']['test_file'] = training_args.output_dir + "/qtc/eval_predictions.json"
+        boolean_config['sn']['output_dir'] = training_args.output_dir+"/sn"
+        boolean_config['sn']['test_file'] = training_args.output_dir + "/evc/eval_predictions.json"
+
+        if model: del model
+        gc.collect()
+        torch.cuda.empty_cache()
+        logger.info(f"torch memory allocated {torch.cuda.memory_allocated()} \
+            max memory {torch.cuda.max_memory_allocated()}")
+
+        cls_main([boolean_config['qtc']])
+        cls_main([boolean_config['evc']])
+        sn_main([boolean_config['sn']])
+
+        with open(os.path.join(boolean_config['sn']['output_dir'], 'eval_predictions_processed.json'), 'r') as f:
+            processed_predictions = json.load(f)
+        references = postprocessor.prepare_examples_as_references(eval_examples)
+        boolean_eval_metric = eval_metrics.compute(predictions=processed_predictions, references=references)
+        path = os.path.join(boolean_config['sn']['output_dir'], f"all_results.json")
+        with open(path, "w") as f:
+            json.dump(boolean_eval_metric, f, indent=4, sort_keys=True)        
+
```

Comment:
Could we move the do_boolean codes into a function i.e. do_boolean(data_args, training_args, model_args, task_config, ...) and let main() call the function?

Reply:
I don't think its necessary yet, but if it gets any bigger I agree.  Probably there should be some coordination with other functionalities in run_mrc.py